pkg/handler/templates.go

```diff
@@ -184,11 +188,14 @@ func (th *TemplateHandler) update(c echo.Context, fillDefaults bool) error {
 	if fillDefaults {
 		tempParams.FillDefaults()
 	}
-	apiTempl, err := th.DaoRegistry.Template.Update(orgID, uuid, tempParams)
+	respTemplate, err := th.DaoRegistry.Template.Update(orgID, uuid, tempParams)
 	if err != nil {
 		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error updating template", err.Error())
 	}
-	return c.JSON(http.StatusOK, apiTempl)
+
+	th.enqueueUpdateTemplateDistributionsEvent(c, orgID, respTemplate.UUID, respTemplate.Date.String(), tempParams.RepositoryUUIDS)
```

Comment:
I think it'd make more sense to lookup the template date from the template at run time.  That way if a user changed it multiple times in rapid succession, or if a task was retried, it'd use whatever the current date is, not what the user set on it when the task was created